internal/autopilot-worker/src/worker.rs

```diff
@@ -0,0 +1,180 @@
+    }
+
+    /// Start the worker and run until cancellation.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the worker fails to start.
+    pub async fn run(&self, cancel_token: CancellationToken) -> Result<()> {
+        let worker = self.executor.start_worker(WorkerOptions::default()).await;
+
+        tokio::select! {
+            () = cancel_token.cancelled() => {
+                tracing::info!("Autopilot worker received shutdown signal");
+                worker.shutdown().await;
+                Ok(())
+            }
+        }
+    }
+}
+
+/// Handle to the running autopilot worker.
+///
+/// This handle exposes the `ToolExecutor` for spawning tool executions
+/// from the gateway.
+#[derive(Clone)]
+pub struct AutopilotWorkerHandle {
+    executor: Arc<ToolExecutor>,
+}
+
+impl AutopilotWorkerHandle {
+    /// Get a reference to the executor for spawning tools.
+    pub fn executor(&self) -> &ToolExecutor {
+        &self.executor
+    }
+}
+
+/// Spawn the autopilot worker as a background task.
+///
+/// The worker will run alongside the gateway and shut down gracefully
+/// when the gateway shuts down.
+///
+/// # Arguments
+///
+/// * `deferred_tasks` - Task tracker from the gateway for spawning background tasks
+/// * `cancel_token` - Cancellation token for graceful shutdown
+/// * `config` - Worker configuration
+///
+/// # Returns
+///
+/// Returns `Some(AutopilotWorkerHandle)` if the worker was successfully spawned,
+/// `None` if spawning failed (error is logged).
+pub fn spawn_autopilot_worker(
+    deferred_tasks: &TaskTracker,
+    cancel_token: CancellationToken,
+    config: AutopilotWorkerConfig,
+) -> Option<AutopilotWorkerHandle> {
+    // We use a oneshot channel to get the handle back from the spawned task
+    let (tx, rx) = tokio::sync::oneshot::channel();
+
+    deferred_tasks.spawn(async move {
+        match AutopilotWorker::new(config).await {
+            Ok(worker) => {
+                tracing::info!("Autopilot worker started");
+                worker.register_tools().await;
+
+                // Create the handle with a shared reference to the executor
+                let handle = AutopilotWorkerHandle {
+                    executor: worker.executor(),
+                };
+
+                // Send the handle back before starting the run loop
+                let _ = tx.send(Some(handle));
+
+                if let Err(e) = worker.run(cancel_token).await {
+                    tracing::error!("Autopilot worker error: {e}");
+                }
+
+                tracing::info!("Autopilot worker stopped");
+            }
+            Err(e) => {
+                tracing::error!("Failed to create autopilot worker: {e}");
+                let _ = tx.send(None);
+            }
+        }
+    });
+
+    // Block briefly to get the handle (this happens during startup)
+    // Using a small timeout to avoid blocking forever if something goes wrong
+    tokio::task::block_in_place(|| {
```

Comment:
I don't think we should do this - if the autopilot takes longer than 30 seconds to start, then it will keep running in the background even though we'll have a shutdown handle of `None`